core/types/transaction.go

```diff
@@ -64,15 +64,54 @@ type Transaction struct {
 	inner TxData    // Consensus contents of a transaction
 	time  time.Time // Time first seen locally (spam avoidance)
 
-	// Arbitrum cache: must be atomically accessed
-	CalldataUnits uint64
+	// Arbitrum cache of the calldata units at a brotli compression level.
+	// The top 8 bits are the brotli compression level last used to compute this,
+	// and the remaining 56 bits are the calldata units at that compression level.
+	calldataUnitsForBrotliCompressionLevel atomic.Uint64
 
 	// caches
 	hash atomic.Pointer[common.Hash]
 	size atomic.Uint64
 	from atomic.Pointer[sigCache]
 }
 
+// GetRawCachedCalldataUnits returns the cached brotli compression level and corresponding calldata units,
+// or (0, 0) if the cache is empty.
+func (tx *Transaction) GetRawCachedCalldataUnits() (uint64, uint64) {
```

Comment:
can we make this function private (lowercase)?

Reply:
No, it's used in nitro to retrieve the cached calldata units if any for a metric update